cmd/abapEnvironmentCheckoutBranch_test.go

```diff
@@ -43,10 +45,185 @@ func TestCheckoutBranchStep(t *testing.T) {
 		err := runAbapEnvironmentCheckoutBranch(&config, nil, &autils, client)
 		assert.NoError(t, err, "Did not expect error")
 	})
+	t.Run("Run Step Failure", func(t *testing.T) {
+		expectedErrorMessage := "Something failed during the checkout: Checking configuration failed: You have not specified any repository configuration to be pulled into the ABAP Environment System. Please make sure that you specified the repositories with their branches that should be pulled either in a dedicated file or via in-line configuration. For more information please read the User documentation"
+
+		var autils = abaputils.AUtilsMock{}
+		defer autils.Cleanup()
+		autils.ReturnedConnectionDetailsHTTP.Password = "password"
+		autils.ReturnedConnectionDetailsHTTP.User = "user"
+		autils.ReturnedConnectionDetailsHTTP.URL = "https://example.com"
+		autils.ReturnedConnectionDetailsHTTP.XCsrfToken = "xcsrftoken"
+
+		config := abapEnvironmentCheckoutBranchOptions{}
+
+		client := &abaputils.ClientMock{
+			BodyList: []string{
+				`{"d" : { "status" : "S" } }`,
+				`{"d" : { "status" : "S" } }`,
+				`{"d" : { "status" : "S" } }`,
+			},
+			Token:      "myToken",
+			StatusCode: 200,
+		}
+
+		err := runAbapEnvironmentCheckoutBranch(&config, nil, &autils, client)
+		assert.EqualError(t, err, expectedErrorMessage)
+	})
+	t.Run("Success case: checkout Branches from file config", func(t *testing.T) {
+		var autils = abaputils.AUtilsMock{}
+		defer autils.Cleanup()
+		autils.ReturnedConnectionDetailsHTTP.Password = "password"
+		autils.ReturnedConnectionDetailsHTTP.User = "user"
+		autils.ReturnedConnectionDetailsHTTP.URL = "https://example.com"
+		autils.ReturnedConnectionDetailsHTTP.XCsrfToken = "xcsrftoken"
+
+		receivedURI := "example.com/Branches"
+		client := &abaputils.ClientMock{
+			Body:       `{"d" : { "__metadata" : { "uri" : "` + receivedURI + `" } } }`,
+			Token:      "myToken",
+			StatusCode: 200,
+		}
+
+		dir, err := ioutil.TempDir("", "test checkout branches")
+		if err != nil {
+			t.Fatal("Failed to create temporary directory")
+		}
+		oldCWD, _ := os.Getwd()
+		_ = os.Chdir(dir)
+		// clean up tmp dir
+
+		defer func() {
+			_ = os.Chdir(oldCWD)
+			_ = os.RemoveAll(dir)
+		}()
+
+		manifestFileString := `
+repositories:
+- name: 'testRepo'
+  branch: 'testBranch'
+- name: 'testRepo2'
+  branch: 'testBranch2'
+- name: 'testRepo3'
+  branch: 'testBranch3'`
+
+		err = ioutil.WriteFile("repositoriesTest.yml", []byte(manifestFileString), 0644)
+
+		config := abapEnvironmentCheckoutBranchOptions{
+			CfAPIEndpoint:     "https://api.endpoint.com",
+			CfOrg:             "testOrg",
+			CfSpace:           "testSpace",
+			CfServiceInstance: "testInstance",
+			CfServiceKeyName:  "testServiceKey",
+			Username:          "testUser",
+			Password:          "testPassword",
+			Repositories:      "repositoriesTest.yml",
+		}
+		err = runAbapEnvironmentCheckoutBranch(&config, nil, &autils, client)
+		assert.NoError(t, err)
```

Comment:
In general, we have many test for different error cases. But the "success" tests could be improved (not just this test). Asserting for "NoError" is not that meaningful. It is not yet tested, if all branches are checked out etc...
But we could improve here on all our steps - maybe this is a topic for a different PR